coordinator/src/main/scala/filodb.coordinator/queryplanner/SingleClusterPlanner.scala

```diff
@@ -185,11 +187,55 @@ class SingleClusterPlanner(val dataset: Dataset,
         .getOrElse(throw new BadQueryException(s"Could not find metric value"))
       val shardValues = shardVals.filterNot(_._1 == dsOptions.metricColumn).map(_._2)
       logger.debug(s"For shardColumns $shardColumns, extracted metric $metric and shard values $shardValues")
-      val shardHash = RecordBuilder.shardKeyHash(shardValues, metric)
-      shardMapperFunc.queryShards(shardHash, spreadProvToUse.spreadFunc(filters).last.spread)
+      val targetSchemaChange = qContext.plannerParams.targetSchema
+        .getOrElse(targetSchemaProvider)
+        .targetSchemaFunc(filters)
+      val targetSchema = if (targetSchemaChange.nonEmpty) targetSchemaChange.last.schema else Seq.empty
+      val shardHash = RecordBuilder.shardKeyHash(shardValues, dsOptions.metricColumn, metric, targetSchema)
+      if(useTargetSchemaForShards) {
+        val nonShardKeyLabelPairs = filters.filter(f => !shardColumns.contains(f.column)
+                                                          && f.filter.isInstanceOf[Filter.Equals])
+                                            .map(cf => cf.column ->
+                                              cf.filter.asInstanceOf[Filter.Equals].value.toString).toMap
+        val partitionHash = RecordBuilder.partitionKeyHash(nonShardKeyLabelPairs, shardVals.toMap, targetSchema,
+          dsOptions.metricColumn, metric)
+        Seq(shardMapperFunc.ingestionShard(shardHash, partitionHash, spreadProvToUse.spreadFunc(filters).last.spread))
```

Comment:
add comment explaining - we go to single shard since all target-schema labels are provided in query filters

Reply:
added documentation